receiver/kafkametricsreceiver/broker_scraper.go

```diff
@@ -40,10 +39,13 @@ func (s *brokerScraper) Name() string {
 	return brokersScraperName
 }
 
-func (s *brokerScraper) start(context.Context, component.Host) error {
+func (s *brokerScraper) setupClient() error {
+	if s.client != nil {
```

Comment:
You also need to update `shutdown` func, otherwise it can panic with nil pointer exception.

Reply:
Good catch. I updated the `shutdown` funcs and added test coverage.